addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -370,5 +407,13 @@ public Integer getResponseCode() {
         public void setResponseCode(Integer responseCode) {
             this.responseCode = responseCode;
         }
+
+        public Map<?, ?> getHeaders() {
+            return headers;
+        }
+
+        public void setHeaders(Map<?, ?> headers) {
+            this.headers = headers;
+        }
```

Comment:
The headers are being handled in a Map in form of key/value pairs, so if we dont restrict headers name/key in GUI then we're expecting a map to hold a value without a key which I think is not possible.

Reply:
It's possible, an empty string is a valid key like any other string.

Reply:
How will it be expressed in YAML file then?
```
headers:
  :"value"
```
Like this? But it is invalid format I believe. Also, no header name and only value is not a valid header and does not seem even to be a valid use case

Reply:
```yaml
headers:
  "":"value"
```

This was already talked in IRC, it doesn't matter if it's a valid header or not, ZAP allows to send malformed headers (to some extent, it should allow even more than it currently does). The automation jobs should not add restrictions to what the underlying functionality supports.

Reply:
Well, in this case I expect the AF headers to nearly always used for submitting real headers, so I'm fine with this restriction 😁 I dont think this job needs to support every single option the underlying code supports, just the most commonly needed ones.

Reply:
I'm thinking on plans to replicate issues (e.g. response splitting), so better if the job(s) don't introduce restrictions.

Reply:
`"":"Value"` works well.

Reply:
Do we still need the restriction to be uplifted from GUI?

Reply:
Done